docs/boilerplate.md

```diff
@@ -0,0 +1,126 @@
+
+### Using Presets
+
+Another source of boilerplate code in any project is the configuration required to get your application up and running. To simplify this process, we've introduced the concept of presets in Oryx.
+
+Presets provide configurations and data structures that are designed to get you up and running quickly without providing a lot of configuration yourself. Presets can be considered as so-called "demo applications", as they typically represent a demo application for a specific business model (e.g. "b2c demo shop"). By using presets, you can quickly configure your application without writing a lot of boilerplate code. This is particularly great when you first start your project or want to demo or test things out.
+
+### Using appBuilder
+
+Finally, we provide an appBuilder function that can be used to set up your application in just a few lines of code. This function takes care of all the boilerplate code for you, including loading presets and configuring your application environment. By using `appBuilder()`, you can avoid writing boilerplate code and focus on building out your application's features.
+
+By using these tactics, we've been able to greatly reduce the amount of boilerplate code required in Oryx projects. This makes it easier to maintain and upgrade your code over time, and allows you to focus on building out your application's features instead of worrying about the underlying framework.
+
+## The minimal boilerplate code
+
+When you create a new project in Oryx, you'll notice that there is a small amount of boilerplate code included by default. This boilerplate code is designed to provide the basic structure and configuration for your application, and can be customized as needed.
+
+The boilerplate contains the following files, with the server files (SSR) being optional.
+
+```
+oryx-app/
+├── app.ts
+├── index.html
+├── package.json
+├── (server/)
+│   ├──render.ts
+│   └──server.ts
+```
+
+Here's a breakdown of the different bits of boilerplate code:
+
+### `package.json`
+
+The package.json file contains all the dependencies of a node js project. To simplify the dependency management, all dependencies are pulled through a single preset package. The preset package contains dependencies to all available oryx packages. This might not be the most optimal setup for your project over time, as it might contain a lot of "dead code", but for the start of your project it is very convenient. While you're getting experienced with Oryx over time, you could consider to build your own list of dependencies to get more fine-grained control. That being said, having unused dependencies in your project will not affect the build time or run time of your project. It is only an overhead during the npm install process.
+
+The bare minimum package.json is setup with the following package.json:
+
+```json
+{
```

Comment:
Maybe it's worth to add a command to run the server?

Reply:
the point was to only show the dependencies of the package.json. I've added a few dots before and after to ensure this is not seen as the full file.